metrics_api/lib/opentelemetry/internal/proxy_instrument.rb

```diff
@@ -5,34 +5,18 @@
 # SPDX-License-Identifier: Apache-2.0
 
 module OpenTelemetry
-  module Internal
-    # @api private
-    class ProxyInstrument
-      def initialize(kind, name, unit, desc, callable)
-        @kind = kind
-        @name = name
-        @unit = unit
-        @desc = desc
-        @callable = callable
-        @delegate = nil
-      end
-
-      def upgrade_with(meter)
-        @delegate = case @kind
-                    when :counter, :histogram, :up_down_counter
-                      meter.send("create_#{@kind}", @name, unit: @unit, description: @desc)
-                    when :observable_counter, :observable_gauge, :observable_up_down_counter
-                      meter.send("create_#{@kind}", @name, unit: @unit, description: @desc, callback: @callback)
-                    end
-      end
-
-      def add(amount, attributes: nil)
-        @delegate&.add(amount, attributes: attributes)
-      end
-
-      def record(amount, attributes: nil)
-        @delegate&.record(amount, attributes: attributes)
-      end
+  module Metrics
+    module ProxyInstrument
     end
   end
 end
+
+require 'opentelemetry/internal/proxy_instrument/delegate_synchronous_instrument'
+require 'opentelemetry/internal/proxy_instrument/counter'
+require 'opentelemetry/internal/proxy_instrument/histogram'
+require 'opentelemetry/internal/proxy_instrument/up_down_counter'
+
+require 'opentelemetry/internal/proxy_instrument/delegate_asynchronous_instrument'
+require 'opentelemetry/internal/proxy_instrument/observable_counter'
+require 'opentelemetry/internal/proxy_instrument/observable_gauge'
+require 'opentelemetry/internal/proxy_instrument/observable_up_down_counter'
```

Comment:
In order to make Proxy Instruments inherit from API Instrument classes, I've created a Proxy Instrument class for each Instrument. They include the `delegate=` setter from a module.